hilti/toolchain/include/ast/attribute.h

```diff
@@ -16,9 +16,60 @@ namespace hilti {
 /** AST node for an attribute. */
 class Attribute : public Node {
 public:
-    /** Returns the name of the attribute, including the leading `&`. */
+    enum Kind {
```

Comment:
I think we definitely want to use an enum class here even though it is more wordy. We not only prefer them due to scoping which prevents namespace pollution, but also since an enum class plays nicer with the type system (classic enums are just integers).